execution.bs

```diff
@@ -3173,67 +3196,91 @@ Insert this section as a new subclause between Header `<stop_token>` synopsis <b
+
+This is the subject of core issue [[cwg2517|2517]].
+</div>
+
+2. Let `t` and `u` be distinct, valid objects of type `T`. The type `T` models `stoppable_token` only if:
 
-    1. All copies of a `stoppable_token` reference the same logical shared stop state and shall report values consistent with each other.
+    1. If `t.stop_possible()` evaluates to `false` then, if `u` references the same logical shared stop state, `u.stop_possible()` shall also subsequently evaluate to `false` and `u.stop_requested()` shall also subsequently evaluate to `false`.
 
-    2. If `t.stop_possible()` evaluates to `false` then, if `u`, references the same logical shared stop state, `u.stop_possible()` shall also subsequently evaluate to `false` and `u.stop_requested()` shall also subsequently evaluate to `false`.
+    2. If `t.stop_requested()` evaluates to `true` then, if `u` references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
 
-    3. If `t.stop_requested()` evaluates to `true` then, if `u`, references the same logical shared stop state, `u.stop_requested()` shall also subsequently evaluate to `true` and `u.stop_possible()` shall also subsequently evaluate to `true`.
+3. Let `t` and `u` be distinct, valid objects of type `T` and let `init` be an
+    object of type `Initializer`. Then for some type `CB`, the type `T` models
+    `stoppable_token_for<CB, Initializer>` only if:
 
-    4. Given a callback-type, CB, and a callback-initializer argument, `init`, of type `Initializer` then constructing an instance, `cb`, of type `T::callback_type<CB>`, passing `t` as the first argument and `init` as the second argument to the constructor, shall,
-        if `t.stop_possible()` is `true`, construct an instance, `callback`, of type `CB`, direct-initialized with `init`, and register callback with `t`’s shared stop state such that callback will be invoked with an empty argument list if a stop request is made on
-        the shared stop state.
+    1. Constructing an instance, `cb`, of type `T::callback_type<CB>`, passing
+        `t` as the first argument and `init` as the second argument to the
+        constructor, shall, if `t.stop_possible()` is `true`, construct an
+        instance, `callback`, of type `CB`, direct-initialized with `init`,
+        and register callback with `t`'s shared stop state such that `callback`
+        will be invoked with an empty argument list if a stop request is made
+        on the shared stop state.
 
-        1. If `t.stop_requested()` is `true` at the time callback is registered then callback may be invoked immediately inline inside the call to `cb`’s constructor.
+        1. If `t.stop_requested()` is `true` at the time `callback` is
+            registered then `callback` may be invoked immediately inline inside
+            the call to `cb`'s constructor.
```

Comment:
Not sure that "invoked immediately inline inside the call" use of "inline" here is normative?

Suggest:
> ... may be invoked on the current thread before call to `cb`'s constructor returns.